palette/src/gradient.rs

```diff
@@ -142,9 +149,14 @@ impl<'a, C: Mix + Clone> Iterator for Take<'a, C> {
 
     fn next(&mut self) -> Option<C> {
         if self.from_head + self.from_end < self.len {
-            let i = self.from + (self.diff / cast(self.len)) * cast(self.from_head);
-            self.from_head += 1;
-            Some(self.gradient.get(i))
+            if self.len == 1 {
+                self.from_head += 1;
+                Some(self.gradient.get(cast(0)))
```

Comment:
Shouldn't this rather be `Some(self.gradient.get(self.from))`? The gradient may be a slice in the middle of the parent gradient.